lib/super_digit.rb

```diff
@@ -1,15 +1,37 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(logn)
+# Space Complexity - O(n) - system stack
 def super_digit(n)
-   
+  sum_digits = sum_digits(n)
+
+  return sum_digits if sum_digits < 10
+
+  return super_digit(sum_digits)
+end
+
+def sum_digits(n)
+  sum = 0
+  while(n > 0)
+    sum += n % 10
+    n /= 10
+  end
+  return sum
 end
-  
 
-# Time Complexity - ?
-# Space Complexity - ?
+
+# Time Complexity - O(logn)
+# Space Complexity - O(n) - system stack
 def refined_super_digit(n, k)
```

Comment:
This works, but it's O( (log nk)^2 ).  You can do better with dynamic programming.